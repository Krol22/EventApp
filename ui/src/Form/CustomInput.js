import React from 'react';

export const CustomInput = (
    {
        field,
        form: {touched, errors},
        label,
        ...props
    }
) => {
    const errorMsg = touched[field.name] && errors[field.name];
    const errClass = errorMsg ? 'error-input' : ''; 

    return (
        <div>
            <div>
                <input {...field} {...props} className={errClass}/>
            </div>
            { errorMsg && (
                <div className="error-msg">
                    * { errorMsg }
                </div>
            )}
        </div>
    );
}