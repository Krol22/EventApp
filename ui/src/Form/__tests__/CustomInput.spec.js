import React from 'react';
import { CustomInput } from '../CustomInput';

import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('CustomInput', () => {

    it('should render properly input', () => {
        const mockField = {
            name: 'testInput',
            type: 'text'
        };

        const mockForm = {
            touched: {
                firstName: true,
            },
            errors: {
                firstName: ''
            }
        };

        const normalInput = shallow(
            <CustomInput field={mockField} form={mockForm} />
        );

        expect(toJson(normalInput)).toMatchSnapshot();
    });

    it('should render input with error when input has errors and was touched', () => {
        const mockForm = {
            touched: {
                testInput: true,
            },
            errors: {
                testInput: 'required'
            }
        };
        const mockField = {
            name: 'testInput',
            type: 'text'
        };
        
        const errorInput = shallow(
            <CustomInput field={mockField} form={mockForm} />
        );

        expect(toJson(errorInput)).toMatchSnapshot();

        expect(errorInput.find('.error-msg').text()).toBe('* required');
    });

});