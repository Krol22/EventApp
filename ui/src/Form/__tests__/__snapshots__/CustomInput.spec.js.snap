// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput should render input with error when input has errors and was touched 1`] = `
<div>
  <div>
    <input
      className="error-input"
      name="testInput"
      type="text"
    />
  </div>
  <div
    className="error-msg"
  >
    * 
    required
  </div>
</div>
`;

exports[`CustomInput should render properly input 1`] = `
<div>
  <div>
    <input
      className=""
      name="testInput"
      type="text"
    />
  </div>
</div>
`;
